### ISSUE TITLE
distance and linkage problem

### ISSUE THREAD

**Issue by finnp** [math-utils/hierarchical-clustering#5]
Hey,

right now the module requires specifying a distance and a linkage function, whereas distance compares two elements. The linkage function then takes all these elements and returns the closest match.

This works well for single, complete and average link functions as cluster distance measures.

However it doesn't work for Centroids or Ward's method, because they compare clusters without individual element distances.

For example the Centroids cluster distance measure calculates the centroid of a cluster of points and then uses this as a measure. There is no element distance required at all.

I'm not sure how to solve this best api-wise. Maybe have a third parameter clusterDistance that could be used instead of `distance` and `linkage`. 

Best,
Finn

**Comment by jonathanong** [comment]
yeah i thought about different methods. it makes the API way too complicated. might be easier just to create another module for those. lol

**Comment by finnp** [comment]
@jonathanong Yeah that's what I thought. The API now simplifies and restricts to those linkage measures. I think that's okay though, since they are the most used anyway.
